### ISSUE TITLE
Is is possible to give .npz files  which is not quantized using gemmaint8?

### ISSUE THREAD

**Issue by musram** [mozilla/firefox-translations-models#173]
It will be nice to give .npz non quantized model so that one can quantize it depending upon his requirements.

**Comment by FricoRico** [comment]
It would be very nice if Firefox actually shared their weights so they could be reused by other frameworks.

**Comment by eu9ene** [comment]
All the weights should be publicly available elsewhere, including intermediate models. The problem is the links to them are not very discoverable. If you're interested in specific languages, you can probably look for successful experiments at https://wandb.ai/moz-translations/projects (the experiment name is a group ID e.g. `experiments-ar-v2_GcMGrkvtQXWSwHMrw85siA`) and then look for the corresponding artifacts in `gs://moz-fx-translations-data--303e-prod-translations-data/models`. Not all the latest models are there, some of them are still on Taskcluster which is also public but again it's hard to find. Or let me know which languages you need and I'll point you to the right place. If it's all languages it will be more difficult.

We're working on automatic uploading to GCS + Model Registry 2.0 which will make any final and intermediate models easily discoverable.

**Comment by FricoRico** [comment]
Thank you so much @eu9ene! And thanks for responding swiftly. I've downloaded a the model to test with. It seems the model structure is slightly different from regular OpusMT/MarianMT models? I'm by no mean an expert and very new to the whole AI stuff.

It seems that the Firefox models have extra RNN layers? The `transformers` package Marian conversion tools do not offer support for this yet and omitting the layers seems to output gibberish (probably because the model is losing its cache between the inference steps). The output is still Korean but with many repeated words.

I attempted to extend the standard tooling to map the RNN layers to PyTorch layers. But I have no clue how the mapping should be.

**Comment by eu9ene** [comment]
Yes, conversion to `transformers` is not supported for the student models, only for the teacher one I guess as it's a pure transformer. We've been exploring this here https://github.com/mozilla/translations/issues/804. You can still compile Marian and run the model with it.

**Comment by FricoRico** [comment]
Ahh yes, that seems to be the issue indeed. I'm working on expanding capabilities of my Android app which currently supports OpusMT models converted to ONNX. Some languages are missing or not performing so I was looking further to see which other options I have rather than training them myself.

But as others have pointed out the Firefox student models do not seem to be able to be converted into ONNX models. I'm still very new to the whole machine translation scene so this is the first time I've even heard of student models. Briding Bergamot with JNI is possible but poses its own challenges right now.

In case you are interested, this is my Android application repository;

Android: https://github.com/FricoRico/Versta.Android
Models: https://github.com/FricoRico/Versta.Models
